src/entry_node.cpp

```diff
@@ -115,4 +117,218 @@ Status EntryNode::createShardedBlob(InferenceEngine::Blob::Ptr& dividedBlob, con
+        return finalStatus;
+
+    for (const auto& pair : inputsInfo) {
+        const auto& name = pair.first;
+        auto networkInput = pair.second;
+        auto it = request->inputs().find(name);
+
+        // Network and request must have the same names of inputs
+        if (it == request->inputs().end()) {
+            std::stringstream ss;
+            ss << "Required input: " << name;
+            const std::string details = ss.str();
+            SPDLOG_DEBUG("Missing input with specific name - {}", details);
+            return Status(StatusCode::INVALID_MISSING_INPUT, details);
+        }
+
+        auto& requestInput = it->second;
+
+        auto status = checkIfShapeValuesNegative(requestInput);
+        if (!status.ok())
+            return status;
+
+        if (requestInput.dtype() == tensorflow::DataType::DT_STRING) {
+            // binary inputs will be validated during conversion to blob
+            SPDLOG_DEBUG("Received request containing binary inputs");
+            status = validateNumberOfBinaryInputShapeDimensions(requestInput);
+            if (!status.ok()) {
+                return status;
+            }
+
+            if (checkBinaryInputBatchSizeMismatch(*networkInput, requestInput)) {
+                std::stringstream ss;
+                ss << "Expected: " << networkInput->getEffectiveShape()[0] << "; Actual: " << requestInput.string_val_size();
+                const std::string details = ss.str();
+                SPDLOG_DEBUG("Invalid batch size - {}", details);
+                return Status(StatusCode::INVALID_BATCH_SIZE, details);
+            }
+            continue;
+        }
+
+        status = validatePrecision(*networkInput, requestInput);
+        if (!status.ok())
+            return status;
+
+        status = validateNumberOfShapeDimensions(*networkInput, requestInput);
+        if (!status.ok())
+            return status;
+
+        if (checkBatchSizeMismatch(*networkInput, requestInput)) {
+            std::stringstream ss;
+            ss << "Expected: " << networkInput->getEffectiveShape()[0] << "; Actual: " << requestInput.tensor_shape().dim(0).size();
+            const std::string details = ss.str();
+            SPDLOG_DEBUG("Invalid batch size - {}", details);
+            return Status(StatusCode::INVALID_BATCH_SIZE, details);
+        }
+
+        if (checkShapeMismatch(*networkInput, requestInput)) {
+            std::stringstream ss;
+            ss << "Expected: " << TensorInfo::shapeToString(networkInput->getEffectiveShape())
+               << "; Actual: " << TensorInfo::tensorShapeToString(requestInput.tensor_shape());
+            const std::string details = ss.str();
+            SPDLOG_DEBUG("Invalid shape - {}", details);
+            return Status(StatusCode::INVALID_SHAPE, details);
+        }
+
+        status = validateTensorContentSize(*networkInput, requestInput);
+        if (!status.ok())
+            return status;
+    }
+    return finalStatus;
+}
+
```

Comment:
In essence - all helper methods - validateX, checkY, are the same as the ones in model instance with the different log to be displayed? It's fine to have it this way for upcoming release but the better solution should be sth like:
predictrequestvalidate_utils.hpp
with those minor checkers and then reuse them in both model instance & entry_node.cpp.

Reply:
OK to have it cleaned up in upcoming MR

Reply:
https://jira.devtools.intel.com/browse/CVS-63112